From f4f0ca05930277253a95c91fbbbd79270f624176 Mon Sep 17 00:00:00 2001
From: Hannu Lyytinen <hannu.lyytinen@unikie.com>
Date: Tue, 21 Feb 2023 15:51:02 +0200
Subject: [PATCH] Input event workaround

Disable backpass events to device due to loopback problem between VMs.
JIRA: HYPR-440

Upstream-Status: Inappropriate [seL4-specific]

Signed-off-by: Hannu Lyytinen <hannu.lyytinen@unikie.com>
---
 drivers/input/input.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/input/input.c b/drivers/input/input.c
--- a/drivers/input/input.c
+++ b/drivers/input/input.c
@@ -352,7 +352,12 @@ static void input_event_dispose(struct input_dev *dev, int disposition,
 				unsigned int type, unsigned int code, int value)
 {
-	if ((disposition & INPUT_PASS_TO_DEVICE) && dev->event)
-		dev->event(dev, type, code, value);
+	if ((disposition & INPUT_PASS_TO_DEVICE) && dev->event) {
+		/*
+		 * WAR: Disable backpass events to device
+		 * due to the problem with loopback
+		 * between VMs. JIRA: HYPR-440
+		 */
+	}
 
 	if (!dev->vals)
 		return;
